Dockerfile

```diff
@@ -14,12 +14,29 @@ WORKDIR /build
 # Copy the code into the container
 ADD . /build
 
-# Build the application
-RUN go mod download
-RUN go build -o main .
+# Build the application, one line to minimize the # of layers
+RUN go mod download && go build -o main .
+
+
+
+
+# Second stage of multi-stage build
+FROM alpine:latest
+
+# Set the port environment variable
+ENV PORT=3000
```

Comment:
this should not be set internally. you should have a default port in the go code. and if a env is passed to the docker image, it will overwrite the default value. look up the docker `-e` tag to see how to pass env vars to docker